x-vector/src/X/Data/Vector/Generic.hs

```diff
@@ -14,12 +22,19 @@ module X.Data.Vector.Generic (
   , mapMaybeM
   , imapMaybeM
 
+  -- * Modifying vectors
+
+  -- ** Transposition
+  , transpose
+
```

Comment:
in that case, having `transpose :: Boxed (Unboxed a) -> Boxed (Unboxed a)` in `Unboxed` sounds like a good plan